### PR TITLE
Fixed y-flip in viewport

### PR DESCRIPTION
**Pull request by spydon**

### CODE REVIEW

File: lib/src/common/viewport_transform.dart [hunk 1]
```diff
@@ -31,31 +31,32 @@ class ViewportTransform {
...
-      gridCorrectedY + -translation.y,
+      gridCorrectedY + translation.y * (yFlip ? 1 : -1),
     );
   }
 
   /// Takes the screen coordinates and return the corresponding world coordinates
-  Vector2 getScreenToWorld(Vector2 argScreen) {
+  Vector2 screenToWorld(Vector2 argScreen) {
     final translationCorrectedX = argScreen.x - translation.x;
     final translationCorrectedY = argScreen.y + translation.y;
```
Comment (MrMuMu) [inline lib/src/common/viewport_transform.dart:56]: flip should be here, imo. We only need to flip the translation effect

Comment (MrMuMu) [reply lib/src/common/viewport_transform.dart:56]: like the following: `final translationCorrectedY = argScreen.y + (yFlip ? -1 : 1) * translation.y;`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG
 
+## [Next]
+ - Refactored the particle system
```
Comment (wolfenrain) [inline CHANGELOG.md:4]: Is this not part of another PR?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+ - Cleaned up the code and remove buffer_utils
```
Comment (wolfenrain) [inline CHANGELOG.md:5]: I dont see this change reflected in the code?

File: CHANGELOG.md (hunk 2 above)
Comment (spydon) [reply CHANGELOG.md:4]: I forgot all the changelog entries for the massive PR I just merged and I was lazy and just added them to this one instead 😄

File: CHANGELOG.md (hunk 3 above)
Comment (spydon) [reply CHANGELOG.md:5]: Same as the other comment

**Pull request merged**